### ISSUE TITLE
Individual tests in IDE do not show Run Debug buttons sometimes

### ISSUE THREAD

**Issue by eliveikis** [matepek/vscode-catch2-test-adapter#215]
**Checklist**

- [X ] The issue is about this extension and NOT about a fork.
- [X ] Check the [known issues](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/documents/support.md#known-issues) list.
- [ X] The latest version of the extension was used.

**Describe the bug**

This is my first time using this extension, so perhaps the actual bug description is more fundamental, but at the very least *most* of the time my tests are not showing the "Run" "Debug" etc buttons on top of the individual tests. They have shown on occasion but then disappear at some point. Log attached below.

**To Reproduce**

1. Write some catch2 tests.
2. Install the extension in vscode.
3. Observe the individual test runner buttons appear.
4. Do stuff  (not sure what, but after some amount of time, the buttons disappear).

Uninstalling and reinstalling the extension causes the buttons to reappear. I will add more info as I discover it.

**Desktop**

- Extension Version: v3.2.12 (latest)
- VS Code Version: latest
- Catch2 / Google Test / DOCTest Version: catch2 v2
- OS Type and Version: windows 10 pro, linux ubuntu 18.04 on wsl2
- Using remote-ssh/docker/wsl?: yes see above

**Log** (optional but recommended)

```
[2020-07-27 10:20:53.679] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog
[2020-07-27 10:20:53.679] [INFO] possibly new suite: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog
[2020-07-27 10:20:53.680] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate
[2020-07-27 10:20:53.680] [INFO] possibly new suite: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate
[2020-07-27 10:20:53.680] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:85:5)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:53.682] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:85:5)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:54.638] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\Debug\tests.pdb
[2020-07-27 10:20:54.638] [INFO] possibly new suite: c:\projects\one-game-sdk\build\tests\Debug\tests.pdb
[2020-07-27 10:20:54.639] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\Debug\tests.exe
[2020-07-27 10:20:54.639] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\Debug\tests.ilk
[2020-07-27 10:20:54.639] [INFO] possibly new suite: c:\projects\one-game-sdk\build\tests\Debug\tests.ilk
[2020-07-27 10:20:54.639] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog
[2020-07-27 10:20:54.640] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\Debug\tests.pdb reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.pdb
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:85:5)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:54.640] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\Debug\tests.ilk reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.ilk
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runMicrotasks (<anonymous>)
	at processTicksAndRejections (internal/process/task_queues.js:85:5)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:54.713] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runNextTicks (internal/process/task_queues.js:58:5)
	at listOnTimeout (internal/timers.js:501:9)
	at processTimers (internal/timers.js:475:7)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:54.715] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:54.766] [INFO] reloadTests Catch2 Promise { { major: 2, _minor: 12, _patch: 2 } } c:\projects\one-game-sdk\build\tests\Debug\tests.exe
[2020-07-27 10:20:54.767] [INFO] discovering tests c:\projects\one-game-sdk\build\tests\Debug\tests.exe [
  '[.],*',
  '--verbosity',
  'high',
  '--list-tests',
  '--use-colour',
  'no'
] c:\projects\one-game-sdk\build\tests\Debug
[2020-07-27 10:20:54.778] [INFO] watcher event: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate
[2020-07-27 10:20:54.817] [INFO] testGrouping {
  testName: 'NEW',
  testNameAsId: 'NEW',
  file: 'C:\\projects\\one-game-sdk\\tests\\unit\\factorial.cpp',
  tags: []
} { groupByExecutable: { label: undefined, description: undefined } };
[2020-07-27 10:20:54.818] [DEBUG] problem under reloading {
  reason: [TypeError: t.update is not a function
  	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:35011
  	at l._createSubtreeAndAddTest (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:81014)
  	at l._reloadFromString (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:34879)
  	at l._reloadChildren (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:37192)
  	at runMicrotasks (<anonymous>)
  	at processTicksAndRejections (internal/process/task_queues.js:85:5)
  	at async c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:82991
  	at async e.ExecutableConfig._recursiveHandleRunnable (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27802)],
  filePath: 'c:\\projects\\one-game-sdk\\build\\tests\\Debug\\tests.exe',
  runnable: l {
    _shared: {
      log: o {
        configSection: 'testMate.cpp.log',
        workspaceFolder: undefined,
        outputChannelName: 'C++ TestMate',
        inspectOptions: [Object],
        includeLocation: false,
        targets: [Array],
        nextInspectOptions: undefined,
        configChangeSubscription: [Object]
      },
      workspaceFolder: { uri: [h], name: 'one-game-sdk', index: 0 },
      loadWithTask: [AsyncFunction],
      sendRetireEvent: [Function],
      sendTestRunEvent: [Function],
      sendTestEvents: [Function],
      executeTask: [Function],
      varToValue: [
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object]
      ],
      rngSeed: null,
      execWatchTimeout: 10000,
      retireDebounceTime: 1000,
      _execRunningTimeout: 600000,
      execParsingTimeout: 5000,
      isNoThrow: false,
      enabledTestListCaching: false,
      enabledStrictPattern: null,
      googleTestTreatGMockWarningAs: 'nothing',
      googleTestGMockVerbose: 'default',
      _execRunningTimeoutChangeEmitter: l {
        _disposed: false,
        _options: undefined,
        _leakageMon: undefined,
        _event: [Function]
      },
      onDidChangeExecRunningTimeout: [Function],
      taskPool: { _maxTaskCount: 1, _runningTaskCount: 0, _waitingTasks: [] }
    },
    _rootSuite: s {
      _shared: {
        log: [o],
        workspaceFolder: [Object],
        loadWithTask: [AsyncFunction],
        sendRetireEvent: [Function],
        sendTestRunEvent: [Function],
        sendTestEvents: [Function],
        executeTask: [Function],
        varToValue: [Array],
        rngSeed: null,
        execWatchTimeout: 10000,
        retireDebounceTime: 1000,
        _execRunningTimeout: 600000,
        execParsingTimeout: 5000,
        isNoThrow: false,
        enabledTestListCaching: false,
        enabledStrictPattern: null,
        googleTestTreatGMockWarningAs: 'nothing',
        googleTestGMockVerbose: 'default',
        _execRunningTimeoutChangeEmitter: [l],
        onDidChangeExecRunningTimeout: [Function],
        taskPool: [Object]
      },
      parent: undefined,
      _label: 'C++ TestMate',
      _descriptionBase: '',
      _tooltipBase: '',
      type: 'suite',
      debuggable: false,
      children: [ [o] ],
      _runningCounter: 0,
      _additionalDesciption: '',
      _additionalTooltip: '',
      _file: null,
      _line: null,
      errored: undefined,
      message: undefined,
      id: '1',
      _executables: [ [Object] ],
      _cancellationTokenSource: { _token: undefined, _parentListener: undefined },
      _runningPromise: Promise { undefined },
      _runningPromiseResolver: [Function]
    },
    properties: {
      name: undefined,
      description: undefined,
      varToValue: [
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object], [Object],
        [Object]
      ],
      path: 'c:\\projects\\one-game-sdk\\build\\tests\\Debug\\tests.exe',
      options: {
        cwd: 'c:\\projects\\one-game-sdk\\build\\tests\\Debug',
        env: [Object]
      },
      _frameworkSpecific: {},
      runTask: { before: [], beforeEach: [], after: [], afterEach: [] },
      parallelizationPool: { _maxTaskCount: 1, _runningTaskCount: 0, _waitingTasks: [] }
    },
    frameworkName: 'Catch2',
    frameworkVersion: Promise { { major: 2, _minor: 12, _patch: 2 } },
    _lastReloadTime: 1595845254509.6472,
    _tests: Set {
      {
        _shared: [Object],
        runnable: [Circular],
        parent: [o],
        type: 'test',
        debuggable: true,
        _label: '⚡️ Unexpected ERROR while parsing',
        _additionalDesciption: '',
        _additionalTooltip: '',
        _skipped: true,
        _tags: [],
        _testDescription: '⚡️ Run me for details ⚡️',
        _typeParam: undefined,
        _valueParam: undefined,
        _file: undefined,
        _line: undefined,
        _reverseRoute: undefined,
        id: '7',
        testNameAsId: '⚡️ Unexpected ERROR while parsing',
        _staticEvent: [Object]
      }
    },
    _catch2Version: { major: 2, _minor: 12, _patch: 2 }
  }
}
[2020-07-27 10:20:55.345] [INFO] reloadTests Catch2 Promise { { major: 2, _minor: 12, _patch: 2 } } c:\projects\one-game-sdk\build\tests\Debug\tests.exe
[2020-07-27 10:20:55.346] [DEBUG] reloadTests was skipped due to mtime c:\projects\one-game-sdk\build\tests\Debug\tests.exe
[2020-07-27 10:20:55.680] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\Debug\tests.pdb reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.pdb
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runNextTicks (internal/process/task_queues.js:58:5)
	at listOnTimeout (internal/timers.js:501:9)
	at processTimers (internal/timers.js:475:7)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:55.681] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\Debug\tests.ilk reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.ilk
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:56.742] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runNextTicks (internal/process/task_queues.js:58:5)
	at listOnTimeout (internal/timers.js:501:9)
	at processTimers (internal/timers.js:475:7)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:56.742] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\tests.dir\Debug\tests.tlog\tests.lastbuildstate
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:57.711] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\Debug\tests.pdb reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.pdb
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at runNextTicks (internal/process/task_queues.js:58:5)
	at listOnTimeout (internal/timers.js:501:9)
	at processTimers (internal/timers.js:475:7)
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
[2020-07-27 10:20:57.712] [DEBUG] _recursiveHandleFile: other error... c:\projects\one-game-sdk\build\tests\Debug\tests.ilk reson: [Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.ilk
	at Object.e.isNativeExecutableAsync (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:1:29751)
	at c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:29426
	at async e.ExecutableConfig._recursiveHandleFile (c:\Users\elive\.vscode\extensions\matepek.vscode-catch2-test-adapter-3.2.12\out\dist\main.js:65:27014)]
```

**Comment by matepek** [comment]
Thats weird indeed. Waiting for more info..

**Comment by eliveikis** [comment]
Unable to reproduce after the reinstall. I'll keep testing.

**Comment by eliveikis** [comment]
@matepek now it is happening again. I still don't have a repo though. It's broken both in WSL2 (ubuntu 18.04) and vscode windows pro 64 bit again. Both IDE's running in windows. At one point I was able to get it working in windows by disabling and re-enabling, but no luck now.

**Comment by eliveikis** [comment]
Aha @matepek maybe some interplay between WSL2 and windows? Now it's working in my windows IDE, but not my WSL ubuntu 18.04 one.

**Comment by eliveikis** [comment]
Definitely still occurring. Might be some interplay between wsl2/ubuntu and windows. I never seem to see the buttons in the IDE when targeting ubuntu in windows (via the Remote WSL extension), but they disappear on Windows. Disabling and re-enabling the extension on windows seems to restore them.

**Comment by matepek** [comment]
I'm almost sure this issue is not originated from this extension. [I'm passing the relevant flags properly](https://github.com/matepek/vscode-catch2-test-adapter/blob/master/src/AbstractTest.ts#L147). I would help but I don't have a Windows with wsl2.

This extension is built upon [Test Explorer Live Share](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-test-explorer-liveshare). At first I would debug that and see that the debug flags of TestSuiteEvent and TestEvent are fine on both sides or not. 

Sorry for your issue. :(

**Comment by eliveikis** [comment]
Fair enough. Is there any more information or things I can try to help? The log seemed to show some serious problems:
`[Error: Not a native executable extension on win32: c:\projects\one-game-sdk\build\tests\Debug\tests.pdb`

**Comment by matepek** [comment]
Try to see the logs of [Test Explorer Live Share](https://marketplace.visualstudio.com/items?itemName=hbenl.vscode-test-explorer-liveshare)

**Comment by matepek** [comment]
That error, is not an error: its a `[DEBUG]`

**Comment by eliveikis** [comment]
@matepek 

Some more info.

1. When the bug occurs - the Test side panel section is empty, but clicking on it to focus it results in it getting populated, and then the Run, Debug etc labels appear on the tests in the text editor.

2. Regarding the Test Explorer Live Share, is that extension required? I did not have Live Share or the Test Explorer Live Share extensions installed, I only had Test Explorer. After installing both Live Share and Test Explorer Live Share, the bug remains, and nothing of interest shows in the Test Explorer Live Share log:
```
[2020-08-14 08:32:21.498] [INFO] Looking for Test Explorer
[2020-08-14 08:32:21.498] [INFO] Test Hub found
[2020-08-14 08:32:21.498] [INFO] Looking for VSLS
[2020-08-14 08:32:21.504] [INFO] VSLS found
[2020-08-14 08:32:21.504] [INFO] Trying to create shared service
[2020-08-14 08:32:21.506] [INFO] Shared service created
[2020-08-14 08:32:21.506] [INFO] Trying to get shared service proxy
[2020-08-14 08:32:21.506] [INFO] Shared service proxy found
```

Also I checked other logs and didn't see anything of interest.

On my Linux IDE window (running in windows), I still have never seen the tests populate there.
